### PR TITLE
Select Yul compiler

### PR DESCRIPTION
**Pull request by yann300**
using https://gist.githubusercontent.com/chriseth/4402022ba55f18bde768fc1f8bcaed29/raw/c5736f40a12051901e81581c1f49bcd514659b53/ERC20.yul

### CODE REVIEW

File: src/app/tabs/compileTab/compileTab.js [hunk 1]
```diff
@@ -46,14 +46,16 @@ class CompileTab {
...
+  }
+
   /**
    * Compile a specific file of the file manager
    * @param {string} target the path to the file to compile
    */
   compileFile (target) {
     if (!target) throw new Error('No target provided for compiliation')
-    if (!/\.sol/.exec(target)) throw new Error(`${target} is not a solidity file. It cannot be compiled with solidity compiler`)
-    // only compile *.sol file.
```
Comment (LianaHus) [inline src/app/tabs/compileTab/compileTab.js]: should we check for .sol or .yul?

Comment (yann300) [reply src/app/tabs/compileTab/compileTab.js]: There is a case (when we load a swarm hash) where the extension is not specified, but the content is still solidity. So it kinda make sense to try compiling it.
That basically why I removed that one.

File: src/app/tabs/compileTab/compileTab.js [hunk 2]
```diff
@@ -46,14 +46,16 @@ class CompileTab {
     this.compiler.setEvmVersion(this.evmVersion)
   }
 
+  setLanguage (lang) {
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compileTab.js]: What litteral can be `lang` ? "Solidity" or "solidity" or "sol" ? You may want to have a comment here : 
```
/**
 * ...
 * @params lang {'solidity' | 'yul' | 'vyper'} ...
 */
setLanguage (lang) 
```

File: src/app/tabs/compileTab/compilerContainer.js [hunk 3]
```diff
@@ -227,6 +238,11 @@ class CompilerContainer {
     this.compileTabLogic.runCompiler()
   }
 
+  onchangeLanguage (event) {
```
Comment (GrandSchtroumpf) [inline src/app/tabs/compileTab/compilerContainer.js]: Can it be "selectLangage" or "changeLangage". This is the action that is triggered on change from select.

File: src/app/tabs/compileTab/compileTab.js (hunk 2 above)
Comment (yann300) [reply src/app/tabs/compileTab/compileTab.js]: that can be `Solidity` or `Yul`.
@chriseth is there other valid input for the `language` property?

File: src/app/tabs/compileTab/compilerContainer.js (hunk 3 above)
Comment (yann300) [reply src/app/tabs/compileTab/compilerContainer.js]: i'd keep it like that for the moment as the format `onSomething` is already used in this file

File: src/app/tabs/compileTab/compileTab.js (hunk 2 above)
Comment (chriseth) [reply src/app/tabs/compileTab/compileTab.js]: Not for now.

**Comment by LianaHus** [comment]
@yann300 when I compile with Yul I have "Only "Solidity" is supported as a language." error. Is this expected behavior?

**Comment by yann300** [comment]
Yes I see, let's do that in a subsequent PR, The error message is i guess clear enough to let this PR in,
and at load, the default compiler is still "Solidity" (is it?)
beside that instead of forcing Solidity, we should just remove Yul from the list.
Please create an issue for that.
@chriseth since which version Yul is accepted as a keyword for `language`?

**Comment by LianaHus** [comment]
ok as you want. but I would not release without this fix. I'll create an Issue for that

**Comment by chriseth** [comment]
@yann300 since version 0.5.7

**Comment by LianaHus** [comment]
https://github.com/ethereum/remix-ide/issues/2100

**Pull request merged**